fix(argo-rollouts): Add traefik.io api group

Kaswob opened the pull request:
Add traefik.io (which is the only supported apiGroup for traefik >3.0) to clusterrole and role of argo-rollouts chart

<!--
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.
-->

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

<!-- Changes are automatically published when merged to `main`. They are not published on branches. -->

DrFaust92 commented:
This requires argo rollouts 1.7.0 as well, which did not release yet

yu-croco commented on charts/argo-rollouts/Chart.yaml:
```diff
@@ -20,3 +20,5 @@ annotations:
   artifacthub.io/changes: |
     - kind: added
       description: Added pod labels for the controller and the dashboard components
```
> I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).

Please remove old one.

> I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)

Also please bump version.

yu-croco commented:
Hi @Kaswob , thank you for the PR.
As @DrFaust92 mentioned earlier, we need to wait for upstream. Once upstream releases v 1.7.0, let's proceed. :)

Kaswob commented:
@yu-croco bumped version and removed old changelog entry. I've also updated AppVersion to 1.7.0 so it will be ready for upstream release. Thanks for the info @DrFaust92

BlowaXD commented on charts/argo-rollouts/templates/controller/clusterrole.yaml:
```diff
@@ -248,6 +248,7 @@ rules:
 # Traefik access needed when using the Traefik provider
 - apiGroups:
   - traefik.containo.us
+  - treafik.io
```
Typo?

Kaswob replied:
Yea, thanks for noticing ;) Fixed

yu-croco commented:
Hi @Kaswob , we release argo-helm for v1.7.0, https://github.com/argoproj/argo-helm/pull/2763 .
Let's resume this PR. 🚀

DrFaust92 commented:
[Kaswob](https://github.com/Kaswob) any movement on this? if not ill try to fork of off you so your commits will get it.

Kaswob commented:
@DrFaust92 @yu-croco bumped version, should be fine now. Sorry for the delay, I've been a bit busy recently and missed it.

yu-croco reviewed (approved):
Thank you for your contribution, LGTM.
